dmoj/cptbox/ptdebug_x86.cpp

```diff
@@ -17,30 +17,33 @@ bool pt_debugger::supports_abi(int abi) {
 uint32_t pt_debugger::seccomp_non_native_arch_list[] = { 0 };
 #endif
 
-void pt_debugger::pre_syscall() {
+bool pt_debugger::pre_syscall() {
     struct iovec iovec;
     iovec.iov_base = &regs;
     iovec.iov_len = sizeof regs;
     if (ptrace(PTRACE_GETREGSET, tid, NT_PRSTATUS, &iovec)) {
         perror("ptrace(PTRACE_GETREGSET)");
         abi_ = PTBOX_ABI_INVALID;
+        return false;
     } else {
         abi_ = PTBOX_ABI_X86;
         regs_changed = false;
+        return true;
     }
 }
 
-void pt_debugger::post_syscall() {
+bool pt_debugger::post_syscall() {
     if (!regs_changed || abi_ == PTBOX_ABI_INVALID)
-        return;
+        return true;
```

Comment:
This is only an error if `abi_ == PTBOX_ABI_INVALID`, isn't it? `!regs_changed` should be fine...

Reply:
Return `true` means it's not going to throw.